src/content/reference/firmware.md

```diff
@@ -5666,70 +5759,57 @@ uint8_t val = 0x45;
 EEPROM.write(addr, val);
 ```
 
-### update()
-This method is similar to `EEPROM.write()` however this method will only write data if the cell contents pointed to by `address` is different to `value`. This method can help prevent unnecessary wear on the EEPROM cells.
+When writing more than 1 byte, prefer `put()` over multiple `write()` since it's fater and it ensures
+consistent data even when power is lost while writing.
 
-`update(address, value)`
-
-`address` is the address (int) of the EERPOM location that needs to be updated
-`value` is the byte data (uint8_t) to write
+### clear()
+Erase all the EEPROM so that all reads will return 255 (hexadecimal 0xFF).
 
 ```C++
 // EXAMPLE USAGE
-// Update a byte value to the second byte of EEPROM
-int addr = 1;
-uint8_t val = 0x45;
-EEPROM.update(addr, val);
+// Reset all EEPROM locations to 0xFF
+EEPROM.clear();
 ```
 
-### get()
-This function will retrieve any object from the EEPROM. Two parameters are needed to call this function. The first is an int containing the address from where the object needs to be read, and the second is the object you would like to read.
-
-`get(address, object)`
-
-`address` is the address (int) of the EERPOM location
-`object` is the object data that would be read
-
-```C++
-// EXAMPLE USAGE
-// Read a custom object from EEPROM addres
-int addr = 10;
-float fValue = 0.00f;
-EEPROM.get(addr, fValue);
+Calling this function pauses processor execution (including code running in interrupts) for 800ms since
+no instructions can be fetched from Flash while the Flash controller is busy erasing both EEPROM
+pages.
 
-struct MyObject{
-float field1;
-byte field2;
-char name[10];
-};
-MyObject myObj;
-EEPROM.get(addr, myObj);
-```
+### hasPendingErase()
+### performPendingErase()
+*These are optional functions for advanced use cases*
 
-### put()
-This function will write any object to the EEPROM. Two parameters are needed to call this function. The first is an int containing the address that is to be written, and the second is the object you would like to write.
+After enough data has been written to fill the first page, the EEPROM emulation will write new data
+to a second page. The first page must be erased before being written again.
 
-`put(address, object)`
+Erasing a page of Flash pauses processor execution (including code running in interrupts) for 500ms since
+no instructions can be fetched from Flash while the Flash controller is busy erasing the EEPROM
+page.
 
-`address` is the address (int) of the EERPOM location to write to
-`object` is the object data to write
+`hasPendingErase()` lets the application developer check if a full EEPROM page needs to be erased.
+When the application determines it is safe to pause processor execution to erase EEPROM it calls
+`performPendingErase()`. You can call this at boot, or when your device is idle if you expect it to
+run without rebooting for a long time.
```

Comment:
There are 2 pages used for EEPROM emulation. One takes 300ms to erase and one takes 500ms to erase. It doesn't matter if you have multi-threading on or off since the processor just stops executing instructions while erasing a page of flash. There's nothing you can do about this on the STM32F205. You can see this for example when you DFU flash. The yellow blinking pauses a few times while the Flash is being erased.

The page erase is handled automatically with the user option to handle it manually. If it does not come across clearly I can wordsmith this section a bit.